examples/dht_node.rs

```diff
@@ -0,0 +1,206 @@
+        return;
+    }
+
+    let (pk, sk) = gen_keypair();
+
+    let local: SocketAddr = "0.0.0.0:33445".parse().unwrap();
+
+    let mut core = Core::new().unwrap();
+    let handle = core.handle();
+
+    // Bind a UDP listener to the socket address.
+    let socket = UdpSocket::bind(&local).unwrap();
+
+    // Create a channel for this socket
+    let (tx, rx) = mpsc::unbounded::<(DhtPacket, SocketAddr)>();
+    let server_obj = Rc::new(RefCell::new(Server::new(tx, pk, sk)));
+
+    // get PK bytes of some "random" bootstrap node (Impyy's)
+    let bootstrap_pk_bytes = FromHex::from_hex(
+        "1D5A5F2F5D6233058BF0259B09622FB40B482E4FA0931EB8FD3AB8E7BF7DAF6F")
+        .unwrap();
+    // create PK from bytes
+    let bootstrap_pk = PublicKey::from_slice(&bootstrap_pk_bytes).unwrap();
+
+                 //"51.15.37.145:33445".parse().unwrap()
+    let saddr: SocketAddr = "198.98.51.198:33445".parse().unwrap();
+    let bootstrap_pn = PackedNode::new(true, saddr, &bootstrap_pk);
+    assert!(server_obj.borrow_mut().kbucket.try_add(&bootstrap_pn));
+
+    let (sink, stream) = UdpFramed::new(socket, DhtCodec).split();
+    // The server task asynchronously iterates over and processes each
+    // incoming packet.
+    let server_obj_c = server_obj.clone();
+    let handler = stream.for_each(move |(packet, addr)| {
+        println!("recv = {:?}", packet.clone());
+        let _ = server_obj_c.borrow_mut().handle_packet((packet, addr));
+        Ok(())
+        })
+        .map_err(|err| {
+            // All tasks must have an `Error` type of `()`. This forces error
+            // handling and helps avoid silencing failures.
+            //
+            error!("packet receive error = {:?}", err);
+            Error::new(ErrorKind::Other, "udp receive error")
+        });
+
+    let writer = rx
+        .map_err(|_| Error::new(ErrorKind::Other, "rx error"))
+        .fold(sink, move |sink, packet| {
+            // println!("send = {:?}", packet.clone());
+            debug!("Send {:?} => {:?}", packet.0, packet.1);
+            sink.send(packet)
+        })
+        // drop sink when rx stream is exhausted
+        .map(|_sink| ())
+    ;
+
+    let server = writer.select(handler)
+        .map(|_| ())
+        .map_err(move |(err, _select_next)| {
+            error!("Processing ended with error: {:?}", err);
+            err
+        });
+
+    // 60 seconds for PingRequests
+    let ping_wakeups = Interval::new(Duration::from_secs(60));
+
+    let server_obj_c = server_obj.clone();
+    let ping_sender = ping_wakeups.for_each(move |()| {
+            // println!("ping wakeup");
+            let kbucket_c = server_obj_c.borrow().kbucket.clone().iter();
+            let server_obj_c_c = server_obj_c.clone();
+
+            type LoopFuture = Box<Future<Item = future::Loop<KbucketIter, KbucketIter>, Error = ()>>;
+
+            let packet_sender = future::loop_fn(kbucket_c, move |mut iter| -> LoopFuture {
```

Comment:
Try to use https://github.com/jgallagher/tokio-chat-example/blob/master/tokio-chat-server/src/main.rs#L86..L105